### PR TITLE
Add default connection stream

### PR DESCRIPTION
**Pull request by rekmarks**
- Add default connection stream
- `initProvider` -> `initializeProvider`

### CODE REVIEW

**Comment by Gudahtt** [comment]
Why include a default stream? I thought asking it to be passed in was an intentional design choice - to keep the provider completely removed from the transport.

**Comment by rekmarks** [comment]
@Gudahtt a default is included for consumers where our injection fails, such as the `script-src: self` CSP in Firefox.

A consumer can still pass in a connection stream if they want, of course.

**Comment by Gudahtt** [comment]
Fair enough; I suppose it's not reasonable to expect consumers to know exactly how to construct a `postMessage` stream for the extension.

File: src/initializeProvider.js [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
 
+module.exports = {
+  initializeProvider,
+  setGlobalProvider,
+}
+
+const getDefaultConnectionStream = () => {
+  return new PostMessageDuplexStream({
+    name: 'metamask-inpage',
+    target: 'metamask-contentscript',
```
Comment (Gudahtt) [inline src/initializeProvider.js:12]: This won't work yet - not until the stream is renamed in the extension too, right?

Comment (rekmarks) [reply src/initializeProvider.js:12]: Yeah, that follows a suggestion by @danfinlay to rename them to something less generic.

I have a draft PR on the extension (MetaMask/metamask-extension#8640) that renames the extension and incorporates the changes in #43. (Of course, we can do a more staged merge on the extension side if we wish, but I reckon we should probably just do a single major version bump of this package.)

**Comment by rekmarks** [comment]
We don't need this.

**Pull request closed without merging**